### PR TITLE
fix: Do not override context when capturing non-error exceptions

### PR DESCRIPTION
**Pull request by kamilogorek**
Resolves https://github.com/getsentry/raven-node/issues/442

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -331,7 +331,8 @@ extend(Raven.prototype, {
       cb = kwargs;
       kwargs = {};
     } else {
-      kwargs = kwargs || {};
+      // Do not use reference, as we'll modify this object later on
```
Comment (MaxBittker) [inline lib/client.js]: why did you need to do this one too? because of line 353?

**Review by MaxBittker — APPROVED** [review]
nice turnaround time on the issue!

left some nits, but i think this looks good! my main concern is putting `JSON.parse(stringify(kwargs))`
on the critical path of all sentry errors - will that cause problems if people are currently putting weird stuff there?

File: test/raven.client.js [hunk 2]
```diff
@@ -385,6 +397,41 @@ describe('raven.Client', function() {
...
+      var old = client.send;
+      var info = {
+        extra: {
+          hello: 'there'
+        }
+      };
+      client.send = function mockSend(kwargs) {
+        client.send = old;
+        kwargs.extra.should.have.property('hello', 'there');
+        kwargs.extra.should.not.equal(info);
```
[the remark this answers is not in the document]
Comment (MaxBittker) [reply test/raven.client.js]: I still think line 430 is a false positive and it should be `should.not.equal(info.extra)`

(this test in its current form passes without your change)

[the remark this answers is not in the document]
Comment (kamilogorek) [reply test/raven.client.js]: You're totally right. Fixed.

File: lib/client.js [hunk 3]
```diff
@@ -367,25 +368,24 @@ extend(Raven.prototype, {
...
+      kwargs = kwargs ? JSON.parse(stringify(kwargs)) : {};
```
[the remark this answers is not in the document]
Comment (kamilogorek) [reply lib/client.js]: I read through the codebase once again and it appears that we should be perfectly fine.

https://github.com/getsentry/raven-node/blob/master/lib/client.js#L549-L562

This middleware is passing just an `err` itself, so we only have to preserve `req` just in case someone wants to use it inside `dataCallback`.

**Comment by kamilogorek** [comment]
Merged manually
